### PR TITLE
Contacts view

### PR DESCRIPTION
**Pull request by LIL717**

### CODE REVIEW

**Comment by chrislim** [comment]
Hmm...I'm getting this error when I try to view people:
CoreData: error: (NSFetchedResultsController) object <PersonIdentifier: 0x1740df2c0> (entity: PersonIdentifier; id: 0xd000000001e80004 <x-coredata://CE0E24EE-E6E2-48EF-B4B2-04EE73EBFBCC/PersonIdentifier/p122> ; data: {
    addressBookIds =     (
        "0xd000000000ac000a <x-coredata://CE0E24EE-E6E2-48EF-B4B2-04EE73EBFBCC/AddressBookId/p43>"
    );
    ceaselessId = "9E3AC393-61C6-4A49-9BCF-1CA61962DF57";
    emails = "<relationship fault: 0x17483abc0 'emails'>";
    favoritedDate = nil;
    firstNames = "<relationship fault: 0x17483ab60 'firstNames'>";
    lastNames = "<relationship fault: 0x17483ac20 'lastNames'>";
    notes = "<relationship fault: 0x17483ac00 'notes'>";
    phoneNumbers = "<relationship fault: 0x17483ac40 'phoneNumbers'>";
    prayerRecords = "<relationship fault: 0x17483ac80 'prayerRecords'>";
    queued = nil;
    removedDate = nil;
    representativeInfo = "0x170a2b280 <x-coredata:///PersonInfo/t7895C9EE-4FA1-4E9C-9778-7CF791B47562141>";
    systemRemovedDate = nil;
}) returned nil value for section name key path 'representativeInfo.primaryLastName.name'. Object will be placed in unnamed section
2015-04-08 11:59:18.568 Ceaseless[16858:4072087] CoreData: error: (NSFetchedResultsController) The fetched object at index 220 has an out of order section name 'NG. Objects must be sorted by section name'

File: Ceaseless/ContactsListsViewController.m [hunk 1]
```diff
@@ -78,6 +75,31 @@ - (void)viewDidLoad {
...
 
+//	if (![[NSUserDefaults standardUserDefaults] objectForKey:kLocalLastAddressBookSyncedDate]
+//		&& (self.ceaselessContacts.syncing == YES)) {
+	if (self.ceaselessContacts.syncing == YES) {
+		self.syncingLabel.hidden = NO;
+		[self.activityIndicator startAnimating];
+		self.tableView.userInteractionEnabled = NO;
+		self.tableView.sectionIndexMinimumDisplayRowCount = INT_MAX;
+
+		NSLog (@"first sync");
```
Comment (chrislim) [inline Ceaseless/ContactsListsViewController.m]: I don't think this will always be the first sync.

File: Ceaseless/ContactsListsViewController.m [hunk 2]
```diff
@@ -78,6 +75,31 @@ - (void)viewDidLoad {
...
+		NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
+
+			//listen for contacts to finish syncing
+		[notificationCenter addObserver: self
+							   selector: @selector (enableTable:)
+								   name: kContactsSyncedNotification
+								 object: nil];
+	} else {
```
Comment (chrislim) [inline Ceaseless/ContactsListsViewController.m]: should this be happening elsewhere? Right now I'm seeing the syncing keep going even after the sync is done. I have to leave the view and return in order for the syncing message to go away. Also it felt kind of odd that i could still scroll around the view while the syncing message was overlaid on top.

**Comment by LIL717** [comment]
this is working properly for me now - I can add a note or update settings while the address book is syncing and the personInfo is no longer missing

**Comment by chrislim** [comment]
Great!

**Pull request merged**